### PR TITLE
Allow for custom page parser function

### PR DESCRIPTION
**Pull request by dkuku**
example parser function and usage:
 
    defmodule Parser do
      def parse(path, content) do
        body =
          content
          |> :binary.split("\nxxx\n")
          |> List.last()
          |> String.upcase()

        attrs = %{path: path, length: String.length(body)}

        {attrs, body}
      end
    end

    defmodule Example do
      use NimblePublisher,
        build: Builder,
        from: "test/fixtures/custom.parser",
        as: :custom,
        page_parser: &Parser.parse/2

      assert hd(@custom).body == "BODY\n"
      assert hd(@custom).attrs == %{path: "test/fixtures/custom.parser", length: 5}
    end

### CODE REVIEW

**Comment by josevalim** [comment]
It is looking great! I think though passing a function can be complicated, because it will only work as long as they are not closures. Maybe passing a module instead works better? 🤔

**Comment by josevalim** [comment]
Also, let’s rename the option to `:parser` or `:file_parser` because I don’t believe we use the term “page” anywhere else. :)

**Comment by dkuku** [comment]
When using a module maybe it will be good to extract the current functionality into a behaviour?

**Comment by josevalim** [comment]
We could but since it is a single function I am also thinking it is n strictly necessary. Another option is to make it an optional callback on :build module, but perhaps it is better to not conflate the two?

File: README.md [hunk 1]
```diff
@@ -54,6 +54,10 @@ Each article in the articles directory must have the format:
 
   * `:earmark_options` - an [`%Earmark.Options{}`](https://hexdocs.pm/earmark/Earmark.Options.html) struct
 
+  * :parser - custom function with arity 2 that gets the file path
```
Comment (josevalim) [inline README.md:57]:
```suggestion
  * `:parser` - custom function with arity 2 that gets the file path
```

**Comment by josevalim** [comment]
I have dropped one minor comment. Then let's change it to a module and we should be good to go. Thanks!

**Comment by dkuku** [comment]
Hi @josevalim. I'm not sure if I understood the part with passing the module correctly.

File: README.md [hunk 2]
```diff
@@ -54,6 +54,9 @@ Each article in the articles directory must have the format:
 
   * `:earmark_options` - an [`%Earmark.Options{}`](https://hexdocs.pm/earmark/Earmark.Options.html) struct
 
+  * :`parser` - custom module with a `parse/2` function that receives the file path
```
Comment (josevalim) [inline README.md:57]:
```suggestion
  * `:parser` - custom module with a `parse/2` function that receives the file path
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**